src/main/java/com/zebrunner/carina/webdriver/DriverHelper.java

```diff
@@ -496,6 +506,192 @@ public void addCookie(Cookie cookie) {
+    }
+
+    public Set<Cookie> getAllCookies() {
+        return getDriver().manage().getCookies();
+    }
+
+    /**
+     * Return a cookie String similar to document.cookie in the browser console
+     * Includes the cookie name and value, and doesn't include the path, domain, etc for each cookie
+     * @return String with all the cookie names and their values
+     */
+    public String getAllCookiesString() {
+        return getAllCookies().stream()
+                .map(cookie -> cookie.getName() + "=" + cookie.getValue() + ";")
+                .collect(Collectors.joining());
+    }
+
+    /**
+     * delete a specific cookie
+     *
+     * @param name the name of the cookie being deleted
+     */
+    public void deleteCookieNamed(String name) {
+        getDriver().manage().deleteCookieNamed(name);
+    }
+
+    /**
+     * delete all cookies in the driver session
+     */
+    public void deleteAllCookies() {
+        getDriver().manage().deleteAllCookies();
+    }
+
+    /**
+     * add an item to local storage
+     *
+     * @param name the name of the item to save
+     * @param value the value of the item to save
+     */
+    public void addToLocalStorage(String name, String value) {
+        ((JavascriptExecutor) getDriver()).executeScript(String.format("window.localStorage.setItem('%s','%s');", name, value));
+    }
+
+    /**
+     * add an item to local storage
+     *
+     * @param name the name of the item to save
+     * @param value the value of the item to save
+     */
+    public void addToLocalStorage(String name, Boolean value) {
+        addToLocalStorage(name, String.valueOf(value));
+    }
+
+    /**
+     * check to see if the item is present in local storage
+     *
+     * @param name the name of the item being stored
+     * @return boolean for if the item is present or not
+     */
+    public boolean isItemPresentInLocalStorage(String name) {
+        return ((JavascriptExecutor) getDriver()).executeScript(String.format("return window.localStorage.getItem('%s');", name)) != null;
+    }
+
+    /**
+     * gets the value of the stored item
+     *
+     * @param name the item that is stored
+     * @return value of the item
+     */
+    public String getValueFromLocalStorage(String name) {
+        return (String) ((JavascriptExecutor) getDriver()).executeScript(String.format("return window.localStorage.getItem('%s');", name));
+    }
+
+    /**
+     * gets the name of a stored item via an index
+     *
+     * @param index the position of where the item is stored
+     * @return name of the stored item
+     */
+    public String getNameFromLocalStorage(int index) {
+        return (String) ((JavascriptExecutor) getDriver()).executeScript(String.format("return window.localStorage.key('%s');", index));
+    }
+
+    /**
+     * remove an item from local storage
+     *
+     * @param name the name of the item to remove
+     */
+    public void removeFromLocalStorage(String name) {
```

Comment:
Maybe we need to provide additional info to the user, for example return Optional with local storage item that we will remove if it exist and Optional.empty if there are no such item

Reply:
We could, although that would require extra network calls (if this is a remote web driver) to first run the script to check for the presence of an item and then again to do the actual removal. My initial thought would be that that should be left up to the user (e.g. if they want that behavior they can check for the existence first and capture the value from the getFromLocalStorage method).